solidity/contracts/token/bridge/EverclearTokenBridge.sol

```diff
@@ -0,0 +1,235 @@
+     * @dev Returns the gas payment quote and the total token amount needed (including fees)
+     * @param _destination The destination domain ID
+     * @param _recipient The recipient address on the destination chain
+     * @param _amount The amount of tokens to transfer
+     * @return quotes Array of quotes containing gas payment and token amount requirements
+     */
+    function quoteTransferRemote(
+        uint32 _destination,
+        bytes32 _recipient,
+        uint256 _amount
+    ) public view override returns (Quote[] memory quotes) {
+        quotes = new Quote[](2);
+        quotes[0] = Quote({
+            token: address(0),
+            amount: _quoteGasPayment(_destination, _recipient, _amount)
+        });
+
+        quotes[1] = Quote({
+            token: address(wrappedToken),
+            amount: _amount + feeParams.fee
+        });
+    }
+
+    /**
+     * @notice Transfers tokens to a remote chain via Everclear's intent system
+     * @dev Creates an Everclear intent for cross-chain transfer. The actual Hyperlane message is sent by Everclear
+     * @param _destination The destination domain ID
+     * @param _recipient The recipient address on the destination chain
+     * @param _amount The amount of tokens to transfer
+     * @return bytes32(0) as the transfer ID (actual ID is managed by Everclear)
+     */
+    function transferRemote(
+        uint32 _destination,
+        bytes32 _recipient,
+        uint256 _amount
+    ) external payable override returns (bytes32) {
+        IEverclearAdapter.FeeParams memory _feeParams = feeParams;
+
+        // Charge sender for the fee
+        _transferFromSender(_amount + _feeParams.fee);
+
+        // Create everclear intent
+        _createIntent(_destination, _recipient, _amount, _feeParams);
+
+        // A hyperlane message will be sent by everclear internally
+        // in a separate transaction. See `EverclearSpokeV3.processIntentQueue`.
+        return bytes32(0);
+    }
+
+    /**
+     * @notice Creates an Everclear intent for cross-chain token transfer
+     * @dev Internal function to handle intent creation with Everclear adapter
+     * @param _destination The destination domain ID
+     * @param _recipient The recipient address on the destination chain
+     * @param _amount The amount of tokens to transfer
+     * @param _feeParams The fee parameters for the intent
+     */
+    function _createIntent(
+        uint32 _destination,
+        bytes32 _recipient,
+        uint256 _amount,
+        IEverclearAdapter.FeeParams memory _feeParams
+    ) internal {
+        bytes32 outputAsset = outputAssets[_destination];
+        require(outputAsset != bytes32(0), "ETB: Output asset not set");
+
+        // Check that the fee params are still valid
+        // This will also revert if the feeParams are not set
+        require(
+            _feeParams.deadline > block.timestamp,
+            "ETB: Fee params deadline expired"
+        );
```

Comment:
should we have some buffer here?

Reply:
Removed. Everclear has does this check already and has no buffer: https://github.com/everclearorg/monorepo/blob/2c256760f338ded02dc58c4dee128135aff1d0e9/packages/contracts/src/contracts/intent/FeeAdapter.sol#L405-L407